cmd/influxd/run/config.go

```diff
@@ -106,3 +116,104 @@ func (c *Config) Validate() error {
 	}
 	return nil
 }
+
+func (c *Config) ApplyEnvOverrides() error {
+	return c.applyEnvOverrides("", reflect.ValueOf(c))
+}
+
+func (c *Config) applyEnvOverrides(prefix string, spec reflect.Value) error {
```

Comment:
On L121, possibly.  Here it's a variable because this function is called recursively on nested sub-config structs.  E.g. `Config.Meta` is called with `META` as the prefix so it looks for `META_DIR`, etc..  A global prefix like `INFLUXDB` makes the variables longer (`INFLUXDB_META_DIR`) but might protect against accidental variables in the env being used here.  

I could go either way on this.